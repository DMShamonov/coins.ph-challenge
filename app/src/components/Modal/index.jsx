// @flow

import * as React from 'react';
import ReactModal from 'react-modal';
import classnames from 'classnames';

import Button from 'components/Button';
import Icon from 'components/Icon';

import type { PropsType } from './types.js.flow';

export default function Modal({
  appElement = document.getElementById('app'),
  contentLabel,
  isOpen = false,
  shouldCloseOnOverlayClick = true,
  className,
  onClose,
  children,
}: PropsType): React.Node {
  return isOpen && (
    <ReactModal
      appElement={appElement}
      isOpen={isOpen}
      onRequestClose={onClose}
      contentLabel={contentLabel}
      shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
      portalClassName={classnames('modal', className)}
      overlayClassName="modal__overlay"
      className="modal__dialog"
      bodyOpenClassName="modal-is-open"
    >
      <Button theme="shell" className="modal__close" onClick={onClose}>
        <Icon theme="close" width={18} title="Close" />
      </Button>
      <div className="modal__content">
        {children}
      </div>
    </ReactModal>
  );
}
