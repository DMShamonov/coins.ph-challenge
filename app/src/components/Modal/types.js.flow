// @flow

import type { Node } from 'react';

export type PropsType = {|
  appElement?: HTMLElement,
  contentLabel: string,
  isOpen?: boolean,
  shouldCloseOnOverlayClick?: boolean,
  className?: string,
  onClose: () => void,
  children: Node,
|}
