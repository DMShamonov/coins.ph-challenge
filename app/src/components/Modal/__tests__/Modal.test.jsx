import React from 'react';
import { shallow } from 'enzyme';

import Modal from '../index';

describe('`Modal` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow((
      <Modal
        contentLabel="__LABEL__"
        onClose={jest.fn()}
      >
        __CONTENT__
      </Modal>
    ))).toMatchSnapshot();
  });

  test('when `isOpen` prop is `true`', () => {
    expect(shallow((
      <Modal
        isOpen
        contentLabel="__LABEL__"
        onClose={jest.fn()}
      >
        __CONTENT__
      </Modal>
    ))).toMatchSnapshot();
  });

  test('when `shouldCloseOnOverlayClick` prop is `true`', () => {
    expect(shallow(((
      <Modal
        shouldCloseOnOverlayClick
        contentLabel="__LABEL__"
        onClose={jest.fn()}
      >
        __CONTENT__
      </Modal>
    )))).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(((
      <Modal
        contentLabel="__LABEL__"
        className="__CLASSNAME__"
        onClose={jest.fn()}
      >
        __CONTENT__
      </Modal>
    )))).toMatchSnapshot();
  });
});
