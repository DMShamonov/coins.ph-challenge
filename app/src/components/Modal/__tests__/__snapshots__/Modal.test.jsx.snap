// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Modal\` component should render correctly when \`isOpen\` prop is \`true\` 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="modal-is-open"
  className="modal__dialog"
  closeTimeoutMS={0}
  contentLabel="__LABEL__"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="modal__overlay"
  parentSelector={[Function]}
  portalClassName="modal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <Button
    className="modal__close"
    onClick={[MockFunction]}
    theme="shell"
  >
    <Icon
      theme="close"
      title="Close"
      width={18}
    />
  </Button>
  <div
    className="modal__content"
  >
    __CONTENT__
  </div>
</Modal>
`;

exports[`\`Modal\` component should render correctly when \`shouldCloseOnOverlayClick\` prop is \`true\` 1`] = `""`;

exports[`\`Modal\` component should render correctly with \`className\` prop 1`] = `""`;

exports[`\`Modal\` component should render correctly with default props 1`] = `""`;
